src/eckit/geo/Domain.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+
+namespace eckit::geo {
+
+
+class Domain {
+public:
+    // -- Exceptions
+    // None
+
+    // -- Constructors
+
+    Domain(double north, double west, double south, double east);
+    Domain();
+    Domain(const Domain&) = default;
+    Domain(Domain&&)      = default;
+
+    // -- Destructor
+
+    virtual ~Domain() = default;
+
+    // -- Convertors
+    // None
+
+    // -- Operators
+
+    Domain& operator=(const Domain&) = default;
+    Domain& operator=(Domain&&)      = default;
+    bool operator==(const Domain&) const;
+    bool operator!=(const Domain& other) const { return !operator==(other); }
+
+    // -- Methods
+
+    double north() const { return north_; }
+    double west() const { return west_; }
+    double south() const { return south_; }
+    double east() const { return east_; }
+
+    bool includesNorthPole() const;
+    bool includesSouthPole() const;
+    bool isPeriodicWestEast() const;
+
+    bool contains(double lat, double lon) const;
```

Comment:
```suggestion
    bool contains(double lon, double lat) const;
```
For consistency with PointLonLat, and we could add another overload for `bool contains(const PointLonLat&) const`

Reply:
Yes you're right, the interface for a Point or one of its variants is better (for the BoundingBox class, this one I deleted for clarity). This class should actually be named BoundBoxLonLat as most of it's logic is specific to that coordinate system, but it is currently this way due the widespread use

Reply:
Done